andromedocs-theme/src/components/sidebar.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import { Link, graphql, useStaticQuery } from "gatsby";
+import { Flex, Box, NavLink } from "theme-ui";
+
+const Sidebar = () => {
+  const { allSidebarYaml } = useStaticQuery(graphql`
+    query {
+      allSidebarYaml {
+        edges {
+          node {
+            label
+            link
+            items {
+              label
+              link
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  console.log(allSidebarYaml);
+
+  return (
+    <Flex
+      as="nav"
+      sx={{
+        backgroundColor: "muted.4",
+        flexDirection: "column",
+        flex: "0 0 15%",
+        padding: 4,
+        "a:link": {
+          textDecoration: "none",
+        },
+        a: {
+          color: "inherit",
+        },
+      }}
+    >
+      {allSidebarYaml.edges.map(({ node: { label, link, items } }, index) => (
+        <NavLink key={index} sx={{}}>
+          <Box
+            sx={{
+              ":hover": {
+                borderRadius: "5px",
+                backgroundColor: "focus",
+              },
+            }}
+          >
+            <Link activeStyle={{ color: "#E31C58" }} to={link}>
+              {label}
+            </Link>
+          </Box>
+          <Flex
+            as="ul"
+            sx={{
+              flexDirection: "column",
+            }}
+          >
+            {items?.map((subitem, index) => (
+              <Box
+                key={index}
+                as="li"
+                sx={{
+                  listStyle: "none",
+                  ":hover": {
+                    borderRadius: "5px",
+                    backgroundColor: "focus",
+                  },
+                }}
+              >
+                <Link activeStyle={{ color: "#E31C58" }} to={subitem.link}>
+                  {subitem.label}
+                </Link>
+              </Box>
+            ))}
+          </Flex>
```

Comment:
This whole part should only appear if you have subitems, right? You can add a conditional rendering (`{items && <Flex>...</Flex>}`) to avoid generating an empty `ul` element :pray:

Reply:
And then you can remove the ? operator in `items?.map` since it'll always exist :rocket: